test/server/cards/03_TWI/units/CaptainTyphoProtectingTheSenator.spec.ts

```diff
@@ -0,0 +1,105 @@
+describe('Captain Typho, Protecting The Senator', function () {
+    integration(function (contextRef) {
+        describe('Captain Typho, Protecting The Senator\'s ability', function () {
+            it('should give Sentinel to a friendly unit', function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        hand: ['captain-typho#protecting-the-senator'],
+                        groundArena: ['consular-security-force']
+                    },
+                    player2: {
+                        groundArena: ['wampa'],
+                    }
+                });
+
+                const { context } = contextRef;
+
+                context.player1.clickCard(context.captainTyphoProtectingTheSenator);
+                expect(context.player1).toBeAbleToSelectExactly([context.captainTyphoProtectingTheSenator, context.consularSecurityForce, context.wampa]);
+                context.player1.clickCard(context.consularSecurityForce);
+
+                expect(context.player2).toBeActivePlayer();
+                context.player2.clickCard(context.wampa);
+                // consular security force is automatically choose because of Sentinel
+                expect(context.consularSecurityForce.damage).toBe(4);
```

Comment:
We need to validate that the sentinel falls off at the end of the phase using `context.moveToNextActionPhase()`